### PR TITLE
Introduce new-style noop adapter.

### PR DESCRIPTION
**Pull request by geeknoid**
We want to auto-generate this adapter, but in the meantime here it is.

**Release note**:

```release-note
```

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/1059)

### CODE REVIEW

File: adapter/noop2/noop.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	"istio.io/mixer/template/checknothing"
+	"istio.io/mixer/template/listentry"
+	"istio.io/mixer/template/logentry"
+	"istio.io/mixer/template/metric"
+	"istio.io/mixer/template/quota"
+	"istio.io/mixer/template/reportnothing"
+)
+
+type (
+	handler struct {
```
Comment (guptasu) [inline adapter/noop2/noop.go]: do you want to add all implemented interfaces here too ?

Comment (guptasu) [inline adapter/noop2/noop.go]: and in the handler below

File: adapter/noop2/noop_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
+// Copyright 2017 Istio Authors.
```
Comment (guptasu) [inline adapter/noop2/noop_test.go]: If we auto generate noop, we also autogenerate _test for it ? It seems we should and we might be able to, looking at the content of the test below.

Comment (geeknoid) [inline adapter/noop2/noop_test.go]: Yep, I added a comment to that effect.

File: adapter/noop2/noop.go (hunk 1 above)
Comment (geeknoid) [inline adapter/noop2/noop.go]: I actually removed everything here. As far as I can tell, putting those definitions there has zero effect on anything.

**Pull request merged**